client/me/account/main.jsx

```diff
@@ -184,10 +187,44 @@ const Account = createReactClass( {
 		this.setState( { userLoginConfirm: event.target.value } );
 	},
 
-	validateUsername() {
+	async validateUsername() {
+		const { translate } = this.props;
 		const username = this.getUserSetting( 'user_login' );
+
 		debug( 'Validating username ' + username );
-		this.props.username.validate( username );
+
+		if ( username === user().get().username ) {
+			this.setState( { validation: false } );
```

Comment:
It's implicitly set when we call `clearUsernameValidation` in `componentWillMount`. While I agree this is sub-optimal, the actual state is initialized via this `formBase` mixin. So I'd opt for getting back to this in case we can get rid of this mixin.

Reply:
Sounds good 👍